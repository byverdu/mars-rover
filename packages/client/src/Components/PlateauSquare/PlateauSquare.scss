.tile {
  &.busy {
    opacity: 1;
    background: #cd5c5c;
  }

  &.will-be-busy {
    opacity: 0.1;

    &.busy {
      opacity: 1;
      background: #cd5c5c;
    }
  }
}
