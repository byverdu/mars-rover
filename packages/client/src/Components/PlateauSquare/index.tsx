import React from 'react';
import './PlateauSquare.scss';

const classnames = require('classnames');

export interface PlateauSquareProps {
  axis: string;
  empty: boolean;
}

const PlateauSquare: React.FC<PlateauSquareProps> = ({axis, empty}) => (
  <div
    className={classnames('tile', {empty: empty, busy: !empty})}
    data-axis={axis}
  />
);

export default PlateauSquare
