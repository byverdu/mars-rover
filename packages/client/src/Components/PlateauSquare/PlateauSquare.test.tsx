import React from 'react';
import PlateauSquare, { PlateauSquareProps } from './index';
import enzyme, { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

const props: PlateauSquareProps = {
  axis: 'x0:y0',
  empty: true,
  willBeBusy: true
};

describe('<PlateauSquare />', () => {
  const wrapper: enzyme.CommonWrapper = mount(<PlateauSquare {...props} />);

  it('should create a snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should know it's position on the grid", () => {
    expect((wrapper.props() as PlateauSquareProps).axis).toEqual('x0:y0');
  });

  it('should know if a Rover is on it', () => {
    expect((wrapper.props() as PlateauSquareProps).empty).toEqual(true);
  });

  it('should set the dataset attribute on the html element', () => {
    expect(
      wrapper.getDOMNode().attributes.getNamedItem('data-axis').value
    ).toEqual('x0:y0');
  });
});
