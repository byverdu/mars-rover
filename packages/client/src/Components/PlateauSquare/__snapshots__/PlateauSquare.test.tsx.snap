// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlateauSquare /> should create a snapshot 1`] = `
<PlateauSquare
  axis="x0:y0"
  empty={true}
  willBeBusy={true}
>
  <div
    className="tile will-be-busy"
    data-axis="x0:y0"
  >
    x0:y0
  </div>
</PlateauSquare>
`;
